src/sentry/models/group.py

```diff
@@ -123,22 +123,22 @@ def from_event_id(self, project, event_id):
         Resolves the 32 character event_id string into
         a Group for which it is found.
         """
-        from sentry import features
+        from sentry.models import EventMapping, Event
+        group_id = None
 
-        if features.has('projects:sample-events', project=self):
-            from sentry.models import EventMapping as Model
-        else:
-            from sentry.models import Event as Model
+        for model in Event, EventMapping:
+            try:
+                group_id = model.objects.filter(
+                    project_id=self.id,
+                    event_id=event_id,
+                ).values_list('group_id', flat=True)[0]
+            except IndexError:
+                # Raise a Group.DoesNotExist here since it makes
```

Comment:
This comment is in a strange place after 6df7f42faa2fb55cd9feb226feb9428cba0339a8

Reply:
Also, shouldn't this early return/break rather than running the query twice after a hit?